submissions/usides/friends-app/js/main.js

```diff
@@ -0,0 +1,149 @@
+
+const searchBtn = document.getElementById('search-btn');
+const rightPane = document.getElementById('right-pane');
+const cardsHolder = document.getElementById('cards-holder');
+const optionGroup = document.getElementById('option-group');
+const sortButtons = document.querySelectorAll('button[data-sort]');
+const filterButtons = document.querySelectorAll('button[data-filter]');
+const nameInput = document.getElementById('name-input');
+const errorNotice = document.getElementById('error-notice');
+const loader = document.getElementById('loader');
+
+let humansOriginState;
+
+const currentState = {
+  humans: undefined,
+  sort: 'a-z',
+  filter: 'none',
+  searchValue: '',
+};
+
+const handleSearcBtn = function () {
+  rightPane.classList.toggle('visible');
+};
+
+const handleInput = function () {
+  currentState.searchValue = this.value.toLowerCase();
+  filterHumans(currentState.filter);
+  sortHumans(currentState.sort);
+  updateCards(currentState.humans);
+};
+
+const handleOptionClick = function (e) {
+  const button = e.target.closest('button');
+  if (!button) return;
+  if (button.dataset.sort) {
+    sortButtons.forEach((button) => button.classList.remove('active'));
+    button.classList.add('active');
+    sortHumans(button.dataset.sort);
+  }
+  if (button.dataset.filter) {
+    if (currentState.filter === button.dataset.filter) {
+      button.classList.toggle('active');
+      filterHumans('none');
+    } else {
+      filterButtons.forEach((button) => button.classList.remove('active'));
+      button.classList.add('active');
+      filterHumans(button.dataset.filter);
+    }
+    sortHumans(currentState.sort);
+  }
+  updateCards(currentState.humans);
+};
+
+const filterHumans = function (filterOption) {
+  const searchedName = humansOriginState.filter((elem) =>
+    elem.name.first.toLowerCase().startsWith(currentState.searchValue),
+  );
+  if (filterOption === 'none') {
+    currentState.humans = searchedName;
+  } else {
+    currentState.humans = searchedName.filter(
+      (elem) => elem.gender === filterOption,
+    );
+  }
+  currentState.filter = filterOption;
+};
+
+const sortByAge = (a, b) => (a.dob.age <= b.dob.age ? 1 : -1);
+const sortByName = (a, b) =>
+  a.name.first.toLowerCase() >= b.name.first.toLowerCase() ? 1 : -1;
+
+const sortHumans = function (sortOption) {
+  switch (sortOption) {
+    case 'a-z':
+      currentState.humans.sort((a, b) => sortByName(a, b));
+      break;
+
+    case 'z-a':
+      currentState.humans.sort((a, b) => sortByName(b, a));
+      break;
+
+    case 'young':
+      currentState.humans.sort((a, b) => sortByAge(b, a));
+      break;
+
+    case 'old':
+      currentState.humans.sort((a, b) => sortByAge(a, b));
+      break;
+  }
+
+  currentState.sort = sortOption;
+};
+
+const updateCards = function (arr) {
+  cardsHolder.innerHTML = '';
+  const fragment = document.createDocumentFragment();
+  arr.forEach((elem) => {
```

Comment:
Use object destructuring here, please

Reply:
Done.

Reply:
Also, you can do it in forEach loop
```suggestion
  arr.forEach(({name, picture, dob, gender}) => {
```

Reply:
Good idea.